domain/ide/workspace/folder_test.go

```diff
@@ -493,82 +521,35 @@ func Test_processResults_ShouldCountSeverityByProduct(t *testing.T) {
 
 	filePath := filepath.Join(f.Path(), "dummy.java")
 	scanData := snyk.ScanData{
-		Product:       product.ProductOpenSource,
-		SeverityCount: make(map[product.Product]snyk.SeverityCount),
+		Product: product.ProductOpenSource,
 		Issues: []snyk.Issue{
 			{Severity: snyk.Critical, Product: product.ProductOpenSource, AffectedFilePath: filePath, AdditionalData: snyk.OssIssueData{Key: util.Result(uuid.NewUUID()).String()}},
 			{Severity: snyk.Critical, Product: product.ProductOpenSource, AffectedFilePath: filePath, AdditionalData: snyk.OssIssueData{Key: util.Result(uuid.NewUUID()).String()}},
+			{Severity: snyk.Critical, IsIgnored: true, Product: product.ProductOpenSource, AffectedFilePath: filePath, AdditionalData: snyk.OssIssueData{Key: util.Result(uuid.NewUUID()).String()}},
 			{Severity: snyk.High, Product: product.ProductOpenSource, AffectedFilePath: filePath, AdditionalData: snyk.OssIssueData{Key: util.Result(uuid.NewUUID()).String()}},
 			{Severity: snyk.High, Product: product.ProductOpenSource, AffectedFilePath: filePath, AdditionalData: snyk.OssIssueData{Key: util.Result(uuid.NewUUID()).String()}},
-			{Severity: snyk.Critical, Product: product.ProductInfrastructureAsCode, AffectedFilePath: filePath, AdditionalData: snyk.IaCIssueData{Key: util.Result(uuid.NewUUID()).String()}},
-			// SeverityCount incremented by ScanData.Product
 		},
 	}
 
 	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
+	engineMock.EXPECT().GetWorkflows().AnyTimes()
 	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(),
 		gomock.Any()).Times(1)
 
 	// Act
 	f.processResults(scanData)
 
 	// Assert
-	require.Equal(t, 2, scanData.SeverityCount[product.ProductOpenSource].Critical)
+	require.NotEmpty(t, scanData.GetSeverityIssueCounts())
+	require.Equal(t, product.ProductOpenSource, scanData.Product)
+	require.Equal(t, 3, scanData.GetSeverityIssueCounts()[snyk.Critical].Total)
+	require.Equal(t, 2, scanData.GetSeverityIssueCounts()[snyk.Critical].Open)
+	require.Equal(t, 1, scanData.GetSeverityIssueCounts()[snyk.Critical].Ignored)
 
 	// wait for async analytics sending
 	time.Sleep(time.Second)
```

Comment:
This should be a channel or done with `assert.Eventually`. Just waiting is not the best solution.